src/zenml/zen_server/zen_server_api.py

```diff
@@ -159,6 +172,44 @@ async def set_secure_headers(request: Request, call_next: Any) -> Any:
     return response
 
 
+@app.middleware("http")
+async def track_last_user_activity(request: Request, call_next: Any) -> Any:
+    """A middleware to track last user activity.
+
+    This middleware checks if the incoming request is a user request and
+    updates the last activity timestamp if it is.
+
+    Args:
+        request: The incoming request object.
+        call_next: A function that will receive the request as a parameter and
+            pass it to the corresponding path operation.
+
+    Returns:
+        The response to the request.
+    """
+    global last_user_activity
+    global last_user_activity_reported
+
+    try:
+        if is_user_request(request):
+            last_user_activity = datetime.now(UTC)
+    except Exception as e:
+        logger.debug(
+            f"An unexpected error occurred while checking user activity: {e}"
+        )
+    if (
+        (datetime.now(UTC) - last_user_activity_reported).total_seconds()
+        > DEFAULT_ZENML_SERVER_REPORT_USER_ACTIVITY_TO_DB_SECONDS
+    ):
+        from zenml.models import ServerSettingsUpdate
+
+        last_user_activity_reported = datetime.now(UTC)
+        zen_store().update_server_settings(
```

Comment:
This means users could use the endpoint to set this value to something in the future, correct? Maybe this should use an internal zen store method like `update_last_user_activity` which only exists on the SQLZenStore?

Reply:
Make sense, will look into

Reply:
Reworked